addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -3173,10 +3194,12 @@ export class OdooEditor extends EventTarget {
         const files = getImageFiles(ev.clipboardData);
         const targetSupportsHtmlContent = isHtmlContentSupported(sel.anchorNode);
         const clipboardHtml = ev.clipboardData.getData('text/html');
+        // if pasting on a link whose label is fully selected, replace it by incoming content
+        this._removeFullySelectedLink();
```

Comment:
I think the two utils function (both used only once) only add more boilerplate to an issue that would otherwise be quite simple. Therefore, I suggest :
```js
// Replace entire link if its label is fully selected.
const selection = this.document.getSelection();
const link = closestElement(selection.anchorNode, 'a', true);
if (link && selection.toString() === const innerText = element.innerText.replace(/\u200B/g, '')) {. 
    const start = leftPos(link);
    // Disable isolatated link.
    if (this._fixLinkMutatedElements && this._fixLinkMutatedElements.link === link) {
        this.resetContenteditableLink();
        this._activateContenteditable();
    }
    link.remove();
    setSelection(...start, ...start, false);
}
```
Besides, it is not clear to me why we need to disable the isolated link. Why is that ? If it is a valid reason then I think it would greatly benefit from a comment.

Reply:
`resetContenteditableLink` clears up `this.fixLinkMutatedElements`. I could do a `delete this.fixLinkMutatedElements` instead, if you prefer.

`activateContenteditable` is necessary to be able to place the cursor somewhere outside the link that is about to be deleted (and to continue editing the document).

Reply:
Ok so the comment could be: `// Exit link isolation since we're removing the link and editing outside of it.`

Reply:
Pushed updated code with suggested changes.